test/jdk/com/sun/net/httpserver/bugs/B6431193.java

```diff
@@ -47,52 +47,45 @@ public static void read (InputStream i) throws IOException {
         i.close();
     }
 
-    /**
-     * @param args
-     */
-    public static void main(String[] args) {
+    public static void main(String[] args) throws IOException {
         class MyHandler implements HttpHandler {
             public void handle(HttpExchange t) throws IOException {
-                InputStream is = t.getRequestBody();
-                read(is);
-                // .. read the request body
+                try (InputStream is = t.getRequestBody();
+                     OutputStream os = t.getResponseBody()) {
+                    read(is);
+                    // .. read the request body
                     String response = "This is the response";
-                t.sendResponseHeaders(200, response.length());
-                OutputStream os = t.getResponseBody();
-                os.write(response.getBytes());
-                os.close();
-                error = Thread.currentThread().isDaemon();
+                    t.sendResponseHeaders(200, response.length());
+                    os.write(response.getBytes());
+                    os.close();
+                    error = Thread.currentThread().isDaemon();
+                }
             }
         }
 
-
-        HttpServer server;
+        InetAddress loopback = InetAddress.getLoopbackAddress();
+        HttpServer server = HttpServer.create(new InetSocketAddress(loopback, 0), 10);
         try {
-            InetAddress loopback = InetAddress.getLoopbackAddress();
-            server = HttpServer.create(new InetSocketAddress(loopback, 0), 10);
-
             server.createContext("/apps", new MyHandler());
             server.setExecutor(null);
             // creates a default executor
-                server.start();
+            server.start();
             int port = server.getAddress().getPort();
-            String s = "http://localhost:"+port+"/apps/foo";
             URL url = URIBuilder.newBuilder()
-                      .scheme("http")
-                      .loopback()
-                      .port(port)
-                      .path("/apps/foo")
-                      .toURL();
+                    .scheme("http")
+                    .loopback()
+                    .port(port)
+                    .path("/apps/foo")
+                    .toURL();
             InputStream is = url.openConnection(Proxy.NO_PROXY).getInputStream();
-            read (is);
-            server.stop(0);
+            read(is);
```

Comment:
should use try-with-resource here too. `read(is)` no longer closes `is`.

Reply:
Good point, looking at it again would it be worthwhile to get rid of the `read` method and just use `is.readAllBytes` in it's place?

Reply:
Depends on how much data the test actually sends. If it's a small amount readAllBytes is fine.

Reply:
I think the only data sent is this string: `This is the response`